File: src/vta/Buffers.py
from m5.params import *
from m5.SimObject import SimObject

class BaseBuffer(SimObject):
    type = "BaseBuffer"
    abstract = True
    cxx_header = "vta/base_buffer.hh"
    cxx_class = "gem5::BaseBuffer"

class InputBuffer(BaseBuffer):
    type = "InputBuffer"
    abstract = False
    cxx_header = "vta/buffers.hh"
    cxx_class = "gem5::InputBuffer"

class WeightBuffer(BaseBuffer):
    type = "WeightBuffer"
    abstract = False
    cxx_header = "vta/buffers.hh"
    cxx_class = "gem5::WeightBuffer"

class OutputBuffer(BaseBuffer):
    type = "OutputBuffer"
    abstract = False
    cxx_header = "vta/buffers.hh"
    cxx_class = "gem5::OutputBuffer"

File: src/vta/buffers.cc
#include "vta/buffers.hh"

#include "base/trace.hh"
#include "debug/BaseVTAFlag.hh"

namespace gem5
{
InputBuffer::InputBuffer(const Params &params) : BaseBuffer(params)
{
    DPRINTF(BaseVTAFlag, "create InputBuffer!\n");
}

WeightBuffer::WeightBuffer(const Params &params) : BaseBuffer(params)
{
    DPRINTF(BaseVTAFlag, "create WeightBuffer!\n");
}

OutputBuffer::OutputBuffer(const Params &params) : BaseBuffer(params)
{
    DPRINTF(BaseVTAFlag, "create OutputBuffer!\n");
}
} // namespace gem5
